flashlib.frc.nt/src/main/java/com/flash3388/frc/nt/NtPidTable.java

```diff
@@ -0,0 +1,91 @@
+
+public class NtPidTable {
+
+    private static final String PID_TABLE_NAME = "Pid";
+    private static final String KP_ENTRY_NAME = "Kp";
+    private static final String KI_ENTRY_NAME = "Ki";
+    private static final String KD_ENTRY_NAME = "Kd";
+    private static final String KF_ENTRY_NAME = "Kf";
+    private static final String PID_INPUT_ENTRY_NAME = "Input";
+    private final NtTable mTable;
+
+    public NtPidTable(double kP, double kI, double kD, double kF, DoubleSupplier processVariableSupplier) {
+        mTable = new NtTable.Builder(PID_TABLE_NAME)
+                .addDoubleEntry(KP_ENTRY_NAME, 0)
+                .addDoubleEntry(KI_ENTRY_NAME, 0)
+                .addDoubleEntry(KD_ENTRY_NAME, 0)
+                .addDoubleEntry(KD_ENTRY_NAME, 0)
+                .addDoubleEntry(KF_ENTRY_NAME, 0)
+                .addDoubleEntry(PID_INPUT_ENTRY_NAME, 0)
+                .build();
+
+        setPIDFValues(kP, kI, kD, kF);
+        startProcessVariableUpdateAction(processVariableSupplier);
+    }
+
+    public static NtPidTable createWithP(double kP, DoubleSupplier processVariableSupplier) {
+        return createWithPD(kP, 0, processVariableSupplier);
+    }
+
+    public static NtPidTable createWithPD(double kP, double kD, DoubleSupplier processVariableSupplier) {
+        return createWithPID(kP, 0, kD, processVariableSupplier);
+    }
+
+    public static NtPidTable createWithPI(double kP, double kI, DoubleSupplier processVariableSupplier) {
+        return createWithPID(kP, kI, 0, processVariableSupplier);
+    }
+
+    public static NtPidTable createWithPID(double kP, double kI, double kD, DoubleSupplier processVariableSupplier) {
+        return new NtPidTable(kP, kI, kD, 0, processVariableSupplier);
+    }
+
+    public PidController createController() {
+        return new PidController(
+                new NtDoubleProperty(mTable.getEntry(KP_ENTRY_NAME)),
+                new NtDoubleProperty(mTable.getEntry(KI_ENTRY_NAME)),
+                new NtDoubleProperty(mTable.getEntry(KD_ENTRY_NAME)),
+                new NtDoubleProperty(mTable.getEntry(KF_ENTRY_NAME)));
+    }
+
+    private void setPIDFValues(double kP, double kI, double kD, double kF) {
+        setKp(kP);
+        setKi(kI);
+        setKd(kD);
+        setKf(kF);
+    }
+
+    private void setKp(double value) {
+        mTable.setAsDouble(KP_ENTRY_NAME, value);
+    }
+
+    private void setKi(double value) {
+        mTable.setAsDouble(KI_ENTRY_NAME, value);
+    }
+
+    private void setKd(double value) {
+        mTable.setAsDouble(KD_ENTRY_NAME, value);
+    }
+
+    private void setKf(double value) {
+        mTable.setAsDouble(KF_ENTRY_NAME, value);
+    }
+
+    private void startProcessVariableUpdateAction(DoubleSupplier processVariableSupplier) {
+        createProcessVariableUpdateAction(processVariableSupplier).start();
+    }
+
+    private Action createProcessVariableUpdateAction(DoubleSupplier processVariableSupplier) {
```

Comment:
I must say, this action building looks weird... `configure()`? `setRunWhenDisabled()`? `save()`? mostly the latter and the first one

Reply:
It's to avoid adding too many methods into `Action` whenever a new configuration field is added. See `Action.setConfiguration`, it's a sugar-syntax for this.

This doesn't help resolve the situation of what to do...

Reply:
It was just a question/complaint, unrelated to this situation

Reply:
And I did change it to what you said so...

Reply:
It was meant to be more a discuss then me telling you what to do. I just wanted to present the options I see.